moran.py

```diff
@@ -18,57 +24,102 @@ def output_players(players, outfilename="players.csv"):
         writer = csv.writer(csvfile)
         writer.writerows(rows)
 
-def build_population(players, weights):
+def build_population(i, j, weights):
+    # players = [s() for s in selected_strategies()]
+    sub_players = players[i], players[j]
     population = []
-    for player, weight in zip(players, weights):
+    for player, weight in zip(sub_players, weights):
         for _ in range(weight):
             population.append(player.clone())
     return population
 
-def run_simulations(players, N=2, turns=100, repetitions=1000, noise=0, outfilename=None):
+def write_winner(outfilename, turns, noise, names_inv, repetitions,
+                 N, i, j, seed=None):
+    """
+    Write the winner of a Moran process to file
+    """
+    if seed:
+        axl.seed(seed)  # Seed the process
+    initial_population = build_population(i, j, [1, N-1])
+
+    s1 = str(players[i].clone())
+    s2 = str(players[j].clone())
+
+    # Pull out just the interaction we need
+    outcomes = dict()
+    for pair in [(s1, s1), (s1, s2), (s2, s1), (s2, s2)]:
```

Comment:
Why do we need all four tuples? Can we get rid of `(s2, s1)`?

In the ApproximateMoran we have:

```
 49                 try:
 50                     match_scores = self.cached_outcomes[player_names].sample()
 51                     scores[i] += match_scores[0]
 52                     scores[j] += match_scores[1]
 53                 except KeyError:
 54                     match_scores = self.cached_outcomes[player_names[::-1]].sample()
 55                     scores[i] += match_scores[1]
 56                     scores[j] += match_scores[0]
 57         self.score_history.append(scores)
```

So that it automatically will take care of that case.

Also, unless I'm missing something we don't actually generate data for `(s2, s1)` (we just have matches for `(s1, s2)`.